usaspending_api/etl/management/commands/load_transactions_in_delta.py

```diff
@@ -807,34 +822,31 @@ def update_transaction_lookup_ids(self):
                 {previous_max_id} + ROW_NUMBER() OVER (
                     ORDER BY all_new_transactions.transaction_unique_id
                 ) AS transaction_id,
-                all_new_transactions.detached_award_procurement_id,
-                all_new_transactions.published_fabs_id,
+                all_new_transactions.is_fpds,
                 all_new_transactions.transaction_unique_id
             FROM (
                 (
                     SELECT
-                        dap.detached_award_procurement_id,
-                        NULL AS published_fabs_id,
+                        TRUE AS is_fpds,
                         -- The transaction loader code will convert this to upper case, so use that version here.
                         ucase(dap.detached_award_proc_unique) AS transaction_unique_id
                     FROM
                          dap_filtered AS dap LEFT JOIN int.transaction_id_lookup AS tidlu ON (
-                            dap.detached_award_procurement_id = tidlu.detached_award_procurement_id
+                            ucase(dap.detached_award_proc_unique) = tidlu.transaction_unique_id
                          )
-                    WHERE tidlu.detached_award_procurement_id IS NULL
+                    WHERE tidlu.transaction_unique_id IS NULL
                 )
                 UNION ALL
                 (
                     SELECT
-                        NULL AS detached_award_procurement_id,
-                        pfabs.published_fabs_id,
+                        FALSE AS is_fpds,
                         -- The transaction loader code will convert this to upper case, so use that version here.
                         ucase(pfabs.afa_generated_unique) AS transaction_unique_id
                     FROM
                         pfabs_filtered AS pfabs LEFT JOIN int.transaction_id_lookup AS tidlu ON (
-                            pfabs.published_fabs_id = tidlu.published_fabs_id
+                            ucase(pfabs.afa_generated_unique) = tidlu.transaction_unique_id
                          )
-                    WHERE tidlu.published_fabs_id IS NULL
+                    WHERE tidlu.transaction_unique_id IS NULL
```

Comment:
I'm pretty sure I don't need that, as I'm doing a left join of each of the (date pre-filtered) source tables on the `transaction_id_lookup` table, so each resulting table should only have the source records from the specified table with existing transactions matched to those records in the `transaction_id_lookup` table, and the new source records from the specified table without matching fields from `transaction_id_lookup`.

The only real difference I can potentially see in adding the `is_fpds` check here is the possibility of having the query run faster, if it can ignore the non-fabs/fpds transactions before trying to do the join.

I guess the possibility of the query running faster is probably worth the additional check, so I'll add it.

Reply:
Yep, you're right -- it picks up the source table records (for new inserts) when there's a "miss" (left join and null) in the target table. I was thinking about the join in the wrong direction. 
Still might be good to have the joined records filtered though.